### ISSUE TITLE
404 when trying to localize to new clusters in Leo

### ISSUE THREAD

**Issue by danqrodney** [DataBiosphere/leonardo#417]
I have created multiple new clusters in Leo this morning (e.g. aou-rw-staging-29/all-of-us), and they appear to initialize correctly, but subsequent attempts to localize fail with a 404... e.g.:

POST /notebooks/aou-test-f1-2116819717/all-of-us/api/localize?async=false HTTP/1.1
api_1                          | Authorization: Bearer XXXXX
api_1                          | Accept: application/json
api_1                          | Content-Type: application/json
api_1                          | User-Agent: Swagger-Codegen/1.0.0/java
api_1                          | 
api_1                          | {"~/workspaces/w0rk/.delocalize.json":"data:application/json;base64,eyJkZXN0aW5hdGlvbiI6ImdzOi8vZmMtNTVmMGUyN2UtMGRkOC00NTUyLTk2NTQtMzQ1YzYwZWIyNjU3L25vdGVib29rcyIsInBhdHRlcm4iOiJcXC5pcHluYiQifQ\u003d\u003d","~/workspaces/w0rk/.all_of_us_config.json":"data:application/json;base64,eyJXT1JLU1BBQ0VfSUQiOiJ3MHJrIiwiQVBJX0hPU1QiOiJhcGktZG90LWxvY2FsaG9zdDo4MDgxIiwiQ0RSX1ZFUlNJT05fQklHUVVFUllfREFUQVNFVCI6InRlc3RfbWVyZ2VfZGVjMjYiLCJXT1JLU1BBQ0VfTkFNRVNQQUNFIjoiYW91LXRlc3QtZjEtMjExNjgxOTcxNyIsIkJVQ0tFVF9OQU1FIjoiZmMtNTVmMGUyN2UtMGRkOC00NTUyLTk2NTQtMzQ1YzYwZWIyNjU3IiwiQ0RSX1ZFUlNJT05fQ0xPVURfUFJPSkVDVCI6ImFsbC1vZi11cy1laHItZGV2In0\u003d"}
api_1                          | --> END POST (636-byte body)
api_1                          | <-- HTTP/1.1 404 Not Found (161ms)

Note that we have *not* enabled jupyterlab in staging. So this appears to be unrelated to that functionality.

**Comment by danqrodney** [comment]
We discovered this was due to the installation of our pyclient library (which brings in jupyter via ipywidgets and was messing with the path that jupyter runs in during startup.)

Leo should make sure it runs the startup scripts in a way that isolates the rest of initialization from their effects as much as possible.

**Comment by danqrodney** [comment]
Now that we've turned off installation of our pyclient library in our startup script, localization / notebook integration is working properly again in AofU. But locally when I turn on jupyterlab, I get a 404 in localization again. I can hit jupyterlab successfully on my cluster ( at https://notebooks.firecloud.org/notebooks/aou-test-f1-2116819717/all-of-us/lab), but localization to the cluster fails, which means our client library won't work. 

Not sure whether this is the same underlying issue as we're running into when trying to install pyclient on startup... but it seems likely.

**Comment by calbach** [comment]
I mentioned this on slack, but the issue appears to be that our initialization script was installing a new jupyter binary via transitive pip install dependency, which took precedence on `PATH` over the existing binary that Leo normally invokes. Fully qualifying the path to the Jupyter binary may resolve this particular issue; though you may want more isolation in general (e.g. subshell) for the user initialization script to prevent future similar problems.

**Comment by rtitle** [comment]
I finally got some time to look at this, and I think I have a fix.

1. When passed JupyterLab as an extension, we were `pip install`-ing it with python 3, which forced the Jupyter runtime to python 3 and broke loading the localize extension. I'm going to fix our extension code to install with pip2, and peg the python version when starting Jupyter so this isn't possible in the future.

2. Even when installing with python 2, it didn't work due to this: https://groups.google.com/forum/#!topic/jupyter/wFo4Aw77Omk
Adding the following to `jupyter_notebook_config.py` resolved the issue:
```
import sys
sys.path.append("/etc/jupyter/custom")
```

3. While working on this I noticed our logging to `~/jupyter.log` isn't working (it's a blank file). Think it's due to some `docker exec` semantics, will fix this as well.

@calbach @danqrodney
